test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -1807,6 +1807,33 @@ def step():
         # TODO: add local/sharded/full state_dict and CPU offloading and rank0
         # interface test here, https://github.com/pytorch/pytorch/issues/97163
 
+    @skip_if_lt_x_gpu(2)
+    def test_state_dict_with_none_tensor_state(self):
+        def _run_test(use_orig_params):
+            model = FSDP(TestDummyModel().cuda(), use_orig_params=use_orig_params)
+            optim = torch.optim.Adam(model.parameters(), lr=1e-2)
+
+            def step():
+                loss = model(model.get_input())
+                loss.backward(loss)
+                optim.step()
+
+            step()
+            original_osd = deepcopy(optim.state_dict())
+            for param_id, state in original_osd["state"].items():
+                # Addd customized value
+                state["value1"] = 2.74
+                state["value2"] = None
+
+            osd = FSDP.optim_state_dict(model, optim, optim_state_dict=original_osd)
+            osd_to_load = FSDP.optim_state_dict_to_load(model, optim, osd )
```

Comment:
nit: I wonder if lint will complain about this one
```suggestion
            osd_to_load = FSDP.optim_state_dict_to_load(model, optim, osd)
```